tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -536,21 +555,26 @@ connect_redirect_tests_common(_In_ const struct bpf_object* object, bool dual_st
 void
 test_common(ADDRESS_FAMILY family, IPPROTO protocol)
 {
+    bpf_object_ptr unique_object;
+    unique_object.reset(nullptr);
+
     _initialize_test_globals();
 
     struct bpf_object* object = nullptr;
     _load_and_attach_ebpf_programs(&object);
 
+    unique_object.reset(object);
+
     _globals.family = family;
     _globals.protocol = protocol;
     socket_family_t socket_family = (family == AF_INET) ? socket_family_t::IPv4 : socket_family_t::IPv6;
     socket_family_t dual_stack_socket_family = (family == AF_INET) ? socket_family_t::Dual : socket_family_t::IPv6;
 
-    connect_redirect_tests_common(object, false, _globals.addresses[socket_family]);
-    connect_redirect_tests_common(object, true, _globals.addresses[dual_stack_socket_family]);
+    connect_redirect_tests_common(unique_object.get(), false, _globals.addresses[socket_family]);
+    connect_redirect_tests_common(unique_object.get(), true, _globals.addresses[dual_stack_socket_family]);
 
     // This should also detach the programs as they are not pinned.
-    bpf_object__close(object);
+    bpf_object__close(unique_object.release());
```

Comment:
If you just delete line 577, it would have the same effect, since the destructor of unique_object would do this, right?

Reply:
I am using the same flow as [PR # 2159 ](https://github.com/microsoft/ebpf-for-windows/pull/2192/files), where the unique_ptr is released at the time of closing. IF we checkout line 538 of [tests/end_to_end/end_to_end.cpp](https://github.com/microsoft/ebpf-for-windows/pull/2192/files#diff-f9ee921992a589d5643886350a4641e33e7902b7643a4983110219db80e660ee), it would work with removing bpf_object__close(unique_object.release()); but that PR still kept it for consistency.